pytorch_lightning/strategies/fully_sharded.py

```diff
@@ -183,12 +224,106 @@ def configure_ddp(self) -> None:
 
         # setup optimizers after fully sharded has wrapped the lightning module
         self.setup_optimizers(self.lightning_module.trainer)
+        self.barrier()
+        self.setup_optimizers(trainer)
```

Comment:
Here is duplicated.
Let's keep the previous ones, the setup_optimizer should happen before model_to_device

Reply:
@edward-io is exactly reproducing the logic here. 
@ananthsub @awaelchli  does move_to_device for FSDP happens before set optimizer as well? Was these part of the duplication necessary? 
The comments above mentioned that `FSDP module.to(device) would first summon all parameters`, but we are not using FSDP.apply or module.to. Am I missing something here?

Reply:
> Here is duplicated. Let's keep the previous ones, the setup_optimizer should happen before model_to_device

oops. sorry, this was a bad merge